### PR TITLE
(fix) Fixed from entry app not properly reloading

### PR DESCRIPTION
**Pull request by icrc-psousa**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [ ] My work includes tests or is validated by existing tests.

## Summary
Sometimes, usually on slow connection, the form-entry-app would't proper reload due to required data for the slot still not updated. This PR fixes the issue by adding required data to the hook that triggers the form-entry-app slot reload.
Some other enhancements were included:
- Allow form to be resubmitted after failed validations
- Styling tweaks

### CODE REVIEW

File: src/group-form-entry-workflow/GroupSessionWorkspace.tsx [hunk 1]
```diff
@@ -179,7 +183,7 @@ const GroupSessionWorkspace = () => {
 
   // 3. Update encounter so that it belongs to the created visit
   useEffect(() => {
-    if (encounter && visit) {
+    if (encounter && encounter.patient?.uuid === visit.patient?.uuid) {
```
Comment (icrc-psousa) [inline src/group-form-entry-workflow/GroupSessionWorkspace.tsx:186]: Avoid trying to update an encounter for the wrong patient

File: src/group-form-entry-workflow/GroupSessionWorkspace.tsx [hunk 2]
```diff
@@ -29,7 +29,8 @@ const WorkflowNavigationButtons = () => {
   const store = useStore(formStore);
   const formState = store[activeFormUuid];
   const navigationDisabled =
-    formState !== "ready" || workflowState !== "EDIT_FORM";
+    (formState !== "ready" || workflowState !== "EDIT_FORM") &&
+    formState !== "readyWithValidationErrors";
```
Comment (icrc-psousa) [inline src/group-form-entry-workflow/GroupSessionWorkspace.tsx:33]: Allow form to be resubmitted after validation errors

Comment (icrc-psousa) [reply src/group-form-entry-workflow/GroupSessionWorkspace.tsx:33]: Hi @ibacher, I've created this PR address some issues with
https://github.com/openmrs/openmrs-esm-fast-data-entry-app/pull/53
hook used as workaround wasn't properly working. Please have a look when you have some time

File: src/group-form-entry-workflow/GroupSessionWorkspace.tsx (hunk 1 above)
Comment (ibacher) [reply src/group-form-entry-workflow/GroupSessionWorkspace.tsx:186]: Won't this throw when visit is null? This should probably be something like:

```
encounter && visit && encounter.patient?.uuid === visit.patient?.uuid
```

**Review by ibacher — APPROVED** [review]
Code-wise, this seems mostly fine. The only thing I'm a little nervous about is the style changes in the same PR as the code changes, since the style changes seem immaterial to the rest of the PR.

**Comment by icrc-psousa** [comment]
> Code-wise, this seems mostly fine. The only thing I'm a little nervous about is the style changes in the same PR as the code changes, since the style changes seem immaterial to the rest of the PR.

Agreed, I've now created two additional pull requests with the non related code.

**Pull request merged**